### ISSUE TITLE
Can't open anymore .gpg files in Nemo

### ISSUE THREAD

**Issue by InfoLibre** [linuxmint/nemo-extensions#511]
### Distribution

Linux Mint 21.3 Cinnamon

### Package version

6.0.2

### Frequency

Always

### Bug description

After upgrading to Linux Mint Cinnamon 21.3, I can’t open anymore my .gpg files in Nemo. Unencrypt menu doesn’t appear anymore after a right click on the .gpg files. Must I reconfigure something or is it a bug?

### Steps to reproduce

Right click on a .gpg file.

### Expected behavior

Unencrypt menu should appear.

**Comment by st4r-w4rs** [comment]
same here!

**Comment by WolfMunroe** [comment]
I'm also unable to decrypt/verify .pgp or .gpg files on the context menu of Nemo now with nemo-seahorse plug-in.  
It is still possible for me to encrypt and sign files of other types via the Nemo context menu, but there's no decryption-related options on existing encrypted files.

**Comment by vajdao** [comment]
I resolved it on my end, by right clicking and .pgp or .gpg file, go into properties, and associate it with gpg by typing 'gpg' (without quotes) in the search box at the bottom, then press on the set as default button.

**Comment by WolfMunroe** [comment]
I used gpg in the terminal to decrypt when I ran into the problem but it is much more convenient to have gpg associated with the file type, and I've done as your instructions now.  

However, that's doesn't do anything to fix the problem with nemo-seahorse not working, it's just bypassing nemo-seahorse entirely.

**Comment by vajdao** [comment]
Okay well thats interesting. So to clarify we are talking about the same issue: when I double clicked on the file, it used to pop up a little window (nemo seahorse) to input my password. Then after upgrading mint to 21.3 this double clicking method stopped working and instead opened gpg/pgp file within xed with all the encrypted text in it. Then I was like, okay lets tell nemo not to use xed, and instead use gpg. Then it was again working on double click, the seahorse window popped up to input my password. I entered my password to decrypt, and the decrypted file perfectly appeared within nemo, using GUI elements only. Nothing is bypassing nemo seahorse I dont know where do u get this from...

**Comment by vajdao** [comment]
Okay well thats interesting. So to clarify we are talking about the same issue: when I double clicked on the encrypted file, it used to pop up a little window (nemo seahorse) to input my password and decrypted it.

Then after upgrading mint to 21.3 this double clicking method stopped working and instead opened gpg/pgp file within xed with all the encrypted text in it.

Then I was like, okay lets tell nemo not to use xed, and instead use gpg. Then it was again working on double click, the seahorse window popped up to input my password. I entered my password to decrypt, and the decrypted file perfectly appeared within nemo, using GUI elements only. Nothing is bypassing nemo seahorse I dont know where do u get this from...

**Comment by WolfMunroe** [comment]
Nemo-seahorse is a plug-in for Nemo that adds Encrypt and Sign options to the Context Menu in Nemo for files that are not encrypted and adds Decrypt and Verify as Context Menu options in Nemo for encrypted files.  

Seahorse is a front-end for gpg, so by associating the encrypted files with gpg as the default action, you're just bypassing the problem with nemo-seahorse plug-in by directly sending the file to gpg to decrypt.  

Even when I used gpg from the terminal it popped up a separate GUI window to input my passphrase, just as it does now that I've associated the files with gpg.  (I found that a little odd to be honest.)  I don't know why it did it that way.  

Point is though that the context menu options for nemo-seahorse plug-in are still not appearing on encrypted files.  

Your ability to decrypt is resolved, it's just not resolved by nemo-seahorse being fixed.

**Comment by vajdao** [comment]
> Nemo-seahorse is a plug-in for Nemo that adds Encrypt and Sign options to the Context Menu in Nemo for files that are not encrypted and adds Decrypt and Verify as Context Menu options in Nemo for encrypted files.

Yes, we know that. I don't recall seeing entries for decrypting an already encrypted file in the right-click context menu. All I remember is there were options for Encryption and Sign when I right-clicked on a non-encrypted file. I remember being surprised by this initially when I first started using this plugin. It seemed odd at first, but then I realized that double-clicking triggered the decryption process, and I accepted that.

<br>

> Seahorse is a front-end for gpg, so by associating the encrypted files with gpg as the default action, you're just bypassing the problem with nemo-seahorse plug-in by directly sending the file to gpg to decrypt.
> Even when I used gpg from the terminal it popped up a separate GUI window to input my passphrase, just as it does now that I've associated the files with gpg. (I found that a little odd to be honest.) I don't know why it did it that way.

I delved into the source code and discovered that Nemo-Seahorse leverages the [GPGME library](https://gnupg.org/software/gpgme/index.html), an API designed to facilitate high-level interactions with GnuPG, enabling seamless integration of cryptographic functionalities into the application.
<br>

> Point is though that the context menu options for nemo-seahorse plug-in are still not appearing on encrypted files.
>Your ability to decrypt is resolved, it's just not resolved by nemo-seahorse being fixed.


Yeah, I understand your perspective. From a quick duct-tape fix context, until it officially gets resolved, it may not matter whether the file association or the plugin itself triggers the `gpg`'s decryption process, as long as the encryption and signing functionalities work as intended. This could serve as a temporary solution until the issue is debugged.

However, it's important to confirm with the development team whether there was ever a "decrypt/verify" entry in the right-click menu when right-clicking on *.gpg or *.pgp files.

**Comment by InfoLibre** [comment]
Yes, there was a decrypt entry in the right-click menu when right-clicking on *.gpg or *.pgp files, but there isn't anymore one since  upgrading to Linux Mint Cinnamon 21.3.

**Comment by InfoLibre** [comment]
This bug is critical! One can't anymore decrypt files.

**Comment by WolfMunroe** [comment]
> This bug is critical! One can't anymore decrypt files with our passwords... It should be solved with high priority.

While the Nemo extension isn't offering a decrypt option so this is still an ongoing bug, you can still decrypt your *.pgp and *.gpg files with gpg within Nemo.  

You can right-click on one of the *.pgp files, go to Properties, go to Open With tab in Properties, and tell it to open with gpg.  Then the default action will be to Open with gpg.  gpg's default behavior on *.pgp or *.gpg is to decrypt.  This is the solution that Vajdao mentioned up-thread, quoted below.

> I ~resolved~ temporary fixed it on my end, by right clicking on any .pgp or .gpg file, then I went into properties, and associated it with gpg by typing 'gpg' (without quotes) in the search box at the bottom, then pressed on the "set as default" button (the middle button).

**Comment by InfoLibre** [comment]
It's more convenient than choosing decrypt in the menu. Could someone add this default association in the next version of Nemo, or this default association could be added when installing nemo-seahorse package?

**Comment by Speretta** [comment]
Above it is said to enter gpg as default action. Instead of using gpg in the default action, you can also use `nemo-seahorse-tool -d`. When overwriting file, the gpg command it doesn't produce any GUI so it can't continue. This may also fix the problem a bit.

**Comment by Atalanttore** [comment]
After opening an encrypted *.pgp file with the `pgp` program in Nemo and entering the correct password, nothing happens. That is very, very bad.
